setup/classes/class.ilSetupAgent.php

```diff
@@ -88,9 +88,7 @@ public function getInstallObjective(Setup\Config $config = null): Setup\Objectiv
                 new Setup\Condition\PHPExtensionLoadedCondition("gd"),
                 $this->getPHPMemoryLimitCondition(),
                 new ilSetupConfigStoredObjective($config),
-                $config->getRegisterNIC()
-                    ? new ilNICKeyRegisteredObjective($config)
-                    : new ilNICKeyStoredObjective($config)
+                new ilNICKeyRegisteredObjective($config)
```

Comment:
@klees $config can be null in case of updating, this now leads to a fatal. I suggest to change this to 
```
new ilNICKeyRegisteredObjective($config ?? new Setup\NullConfig())
```
 here

Reply:
not sure if something else is then needed inside `ilNICKeyRegisteredObjective`

Reply:
@lukastocker Please look into this.

Reply:
> not sure if something else is then needed inside `ilNICKeyRegisteredObjective`



Hi @chfsx ,

to be sure: the $config can't be null for the installation - only for the update process, correct?

Reply:
https://github.com/ILIAS-eLearning/ILIAS/pull/7259

Reply:
Thanks a lot, es this seems to be the better solution in the PR. and yes, config can only be null during update